app/services/concerns/xccdf/hosts.rb

```diff
@@ -35,14 +35,12 @@ def external_report?
     private
 
     def test_result_profile
-      ::Profile.canonical
-               .where(ref_id: @test_result_file.test_result.profile_id,
-                      benchmark: benchmark).first ||
-        ::Profile.find_or_initialize_by(
-          ref_id: @test_result_file.test_result.profile_id,
-          name: @test_result_file.test_result.profile_id,
-          benchmark_id: benchmark.id
-        )
+      @test_result_profile ||= ::Profile.canonical.find_or_initialize_by(
+        ref_id: @test_result_file.test_result.profile_id,
+        benchmark: benchmark
+      ).tap do |profile|
+        profile.name ||= @test_result_file.test_result.profile_id
```

Comment:
Nitpick: Could have used `create_with(name: @test_result_file.test_result.profile_id)` (before `find_or_initialize_by`) to avoid setting name on found profile.